test/cmd/discovery.sh

```diff
@@ -407,3 +407,31 @@ __EOF__
   set +o nounset
   set +o errexit
 }
+
+run_explain_crd_with_additional_properties_tests() {
+  set -o nounset
+  set -o errexit
+
+  # create_and_use_new_namespace
+  kube::log::status "Testing explain with custom CRD that uses additionalProperties as non boolean field"
+
+  output_message=$(kubectl apply -f hack/testdata/CRD/example-crd-with-additionalfields.yaml)
+  kube::test::if_has_string "${output_message}" 'created'
+
+  kube::test::wait_object_assert customresourcedefinitions "{{range.items}}{{if eq ${id_field:?} \"mock-resources.test.com\"}}{{$id_field}}:{{end}}{{end}}" 'mock-resources.test.com:'
+
+  # For some reason without this explain fails with message no resource found.
+  sleep 1
```

Comment:
I couldn't find any solution for this do you have any idea?
Already used `ube::test::wait_object_assert`. Keeps failing if i don't use sleep here.

Reply:
Instead of sleep, what about sending dummy kubectl explain request to aggregated discovery populate the local cache by ignoring the output and sending another actual `kubectl explain mock-resource`?. I'd use alternative way other than sleep.

Reply:
I tried some, it seems we have to make it wait for 1 second either by running few `kubectl explain` or using `sleep` i gradually increase the number of `kubectl explain` and we have to almost use 7 of them so it can be replaced with `sleep`. If there is another way i'm happy to implement but i think using `sleep 1` is more cleaner and readable than using 7 of `kubectl explain po --recursive`.
What do you think?

Reply:
How do we know that this sleep won't be sufficient and not create any flakiness in tests?. We have to deterministic in this behavior to prevent any such intermittent test failures. If we figure out, `sleep 1` is the deterministic behavior due to a reason, I'm fine pursuing with that.

Reply:
I was thinking about this, either way flakiness would be there if we use `kubectl explain` few times as well.
Maybe i can create a retry like `wait_object_assert` for this behavior.

Reply:
> I was thinking about this, either way flakiness would be there if we use `kubectl explain` few times as well. Maybe i can create a retry like `wait_object_assert` for this behavior.

That would work and would be far better.